File: 3483-alternating-groups-ii/alternating-groups-ii.cpp
class Solution {
public:
    int numberOfAlternatingGroups(vector<int>& colors, int k) {

        k--;

        int j = 0;

        while(j < k) colors.push_back(colors[j++]);

        int ans = 0;
        j = 0;

        for(int i = 1; i < colors.size(); i++) {

            if(colors[i] == colors[i - 1]) j = i;
            
            else if(i - j == k) {
                ans++;
                j++;
            }
        }

        return ans;
        
    }
};

File: 3483-alternating-groups-ii/alternating-groups-ii.js
/**
 * @param {number[]} colors
 * @param {number} k
 * @return {number}
 */
var numberOfAlternatingGroups = function(colors, k) {



    let j = 0, ans = 0;
    k--;

    while(j < k) {
        colors.push(colors[j++]);
    }
    j = 0;
    
    let n = colors.length


    
    for(let i = 1; i < n; i++) {

        if(colors[i] == colors[i - 1]) j = i;

        else if(i - j == k) {

            ans++;
            j++;
        }
    }

    
    return ans;
    
};